watchtower-plugin/src/dbm.rs

```diff
@@ -139,24 +140,25 @@ impl DBM {
         .map_err(|_| Error::NotFound)
     }
 
-    /// Stores a tower record into the database.
+    /// Stores a tower record into the database alongside the corresponding registration receipt.
     pub fn store_tower_record(
-        &self,
+        &mut self,
         tower_id: TowerId,
         net_addr: &str,
         receipt: &RegistrationReceipt,
     ) -> Result<(), Error> {
-        let query =
-            "INSERT OR REPLACE INTO towers (tower_id, net_addr, available_slots, subscription_expiry) VALUES (?1, ?2, ?3, ?4)";
-        self.store_data(
-            query,
-            params![
-                tower_id.to_vec(),
-                net_addr,
-                receipt.available_slots(),
-                receipt.subscription_expiry()
-            ],
+        let tx = self.get_mut_connection().transaction().unwrap();
+        tx.execute(
+            "INSERT INTO towers (tower_id, net_addr, available_slots) VALUES (?1, ?2, ?3) ON CONFLICT (tower_id) DO UPDATE SET net_addr = ?2, available_slots = ?3",
```

Comment:
Shouldn't the slots be added to the already existing ones?
Realized this while hardening `test_store_load_towers` test: I needed to store only the receipt with the greatest sub expiry but the tower's `available_slots` was messed up because we update it with `store_tower_record` using random receipts.

Reply:
That may be due to how the test helpers are built. In theory, when receiving a new registration receipt the tower already adds either (or both) the slots and time to the current one.

If we update it with a new random receipt, this won't hold given the receipt is not tied to the previous one in this case.